CMake fails looking for GLEW::glew

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Trying to run cmake on debian bookworm, it tells me it can't find `GLEW::glew` when using VC_PKG

### Steps to Reproduce

Run `cmake .. -DCMAKE_BUILD_TYPE=Debug -DES_USE_VCPKG=1


### Expected Behavior

cmake finishes.

### Screenshots

```
CMake Error at CMakeLists.txt:173 (target_link_libraries):
  The link interface of target "ExternalLibraries" contains:

    GLEW::glew

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

### Link to save file

_No response_

### Operating System

debian bullseye

### Game Source

Built from source

### Game Version

v0.10.9

### Additional Information

Running with `-DES_USE_VCPKG=0` works fine. Its only when using VCPKG that it breaks.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,7 +211,12 @@ else()
 		target_compile_definitions(EndlessSkyLib PUBLIC GL_SILENCE_DEPRECATION)
 	else()
 		# GLEW is only needed on Linux and Windows.
-		target_link_libraries(ExternalLibraries INTERFACE GLEW::glew)
+		if(TARGET GLEW::GLEW)
+			set(GLEW_TARGET_NAME GLEW::GLEW)
+		else()
+			set(GLEW_TARGET_NAME GLEW::glew)
+		endif()
+		target_link_libraries(ExternalLibraries INTERFACE ${GLEW_TARGET_NAME})
 	endif()
 endif()
 

